static/app/views/performance/trends/changeExplorerUtils/metricsChart.tsx

```diff
@@ -0,0 +1,468 @@
+import {ReactNode, ReactText, useMemo} from 'react';
+import {Location} from 'history';
+import moment from 'moment';
+
+import GridEditable, {
+  COL_WIDTH_UNDEFINED,
+  GridColumnOrder,
+} from 'sentry/components/gridEditable';
+import SortLink from 'sentry/components/gridEditable/sortLink';
+import {t} from 'sentry/locale';
+import {Organization} from 'sentry/types';
+import {parsePeriodToHours} from 'sentry/utils/dates';
+import {TableData, useDiscoverQuery} from 'sentry/utils/discover/discoverQuery';
+import EventView from 'sentry/utils/discover/eventView';
+import {
+  AggregationKeyWithAlias,
+  ColumnType,
+  fieldAlignment,
+  QueryFieldValue,
+} from 'sentry/utils/discover/fields';
+import {Container} from 'sentry/utils/discover/styles';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {formatPercentage} from 'sentry/utils/formatters';
+import theme from 'sentry/utils/theme';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import {TransactionThresholdMetric} from 'sentry/views/performance/transactionSummary/transactionThresholdModal';
+import {ExplorerText} from 'sentry/views/performance/trends/changeExplorer';
+import {
+  NormalizedTrendsTransaction,
+  TrendFunctionField,
+  TrendsTransaction,
+  TrendView,
+} from 'sentry/views/performance/trends/types';
+
+type MetricsChartProps = {
+  isLoading: boolean;
+  location: Location;
+  organization: Organization;
+  transaction: NormalizedTrendsTransaction;
+  trendFunction: string;
+  trendView: TrendView;
+};
+
+export function MetricsChart(props: MetricsChartProps) {
+  const {trendFunction, transaction, trendView, organization, location, isLoading} =
+    props;
+  let p50: TableDataRow[] = [];
+  let p95: TableDataRow[] = [];
+  let fieldsNeeded: AggregationKeyWithAlias[] = useMemo(() => ['tps'], []);
+
+  if (trendFunction === TrendFunctionField.P50) {
+    p50 = getTrendsRowData(transaction, TrendFunctionField.P50);
+  }
+
+  fieldsNeeded = useMemo(
+    () =>
+      trendFunction !== TrendFunctionField.P50
+        ? [...fieldsNeeded, 'p50']
+        : [...fieldsNeeded],
+    [fieldsNeeded, trendFunction]
+  );
+
+  if (trendFunction === TrendFunctionField.P95) {
+    p95 = getTrendsRowData(transaction, TrendFunctionField.P95);
+  }
+
+  fieldsNeeded = useMemo(
+    () =>
+      trendFunction !== TrendFunctionField.P95
+        ? [...fieldsNeeded, 'p95']
+        : [...fieldsNeeded],
+    [fieldsNeeded, trendFunction]
+  );
+
```

Comment:
This needs to be cleaned up, the dep arrays referring to the variable they're memoizing to create an array, I'm not sure this even needs to be memoized, I'd have to walk through it after pulling the code. The end result is just an array of `['p50','p95','eps']` every time since it's required by the table, no?